realm/realm-jni/src/io_realm_internal_tableview.cpp

```diff
@@ -1015,3 +1015,17 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableView_nativeSync(
     } CATCH_STD()
     return 0;
 }
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_TableView_nativeFindBySourceNdx
+        (JNIEnv *env, jobject, jlong nativeViewPtr, jlong rowIndex)
+{
+    TR_ENTER_PTR(nativeViewPtr);
+    try {
+        if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr) || !ROW_INDEX_VALID(env, TV(nativeViewPtr), rowIndex))
```

Comment:
The problem is `ROW_INDEX_VALID(env, TV(nativeViewPtr), rowIndex)` is checking if a row from parent table is valid in the table view. rowIndex (should be clear if name it as `sourceRowIndex`) is the index of the row in the table. But here, the `nativeViewPtr` is the pointer of the `TableView`.

I think it is not necessary to call this check. Assertion would happen if the `rowIndex` > `Tabel.size` in core.